src/process/exec.h

```diff
@@ -55,12 +31,14 @@ static __always_inline void exit_exec(struct pt_regs *ctx, process_message_type_
     int retcode = (int)PT_REGS_RC(ctx);
     if (retcode < 0) return;
 
+    pm->u.syscall_info.data.exec_info.event_id = push_scripts(ctx, buffer);
+
     void *ts = (void *)bpf_get_current_task();
     u64 pid_tgid = bpf_get_current_pid_tgid();
 
     // do not emit if we couldn't fill the syscall info
     ret = fill_syscall(&pm->u.syscall_info, ts, pid_tgid >> 32);
-    if (ret > 0) return;
+    if (ret > 0) return; // TODO: send discard event if pm->u.syscall_info.data.exec_info.event_id != 0?
```

Comment:
What's the idea behind these "send discard event?" comments?  Have we already sent something that we later need to ignore?

Reply:
At this point we have already sent the script information (if there was any to send) which the userspace would cache. I wonder if we should send discard events here or just let user-space age those out through some form of LRU cache or something. 

I don't think it's a big deal because we only send script events after we check the retcode of exec so the only things that could fail are if the exec was a "kernel process" which I don't think we get for execs for (much less execs with scripts) or errors in the ebpf program (e.g., missing offsets) which so far I have only seen happen during shutdown of a program. 

The TODO in here is mostly so I know "we'll probably want to address this later just to make it more fool-proof".